### PR TITLE
docs(dinghy): DOC-327 add appmetadata field (#490)

### PR DESCRIPTION
**Pull request by 317brian**
(cherry picked from commit 2940fdf69e1ab5a23325ad90689cef5b04f7ae32)

Resolves GitHub issue: <link to GH issue>

Resolves Jira: [JIRAPROJECT-123]

### CODE REVIEW

**Comment by 317brian** [comment]
Don't merge for yet. There is a bug w/ the feature.

**Comment by 317brian** [comment]
This is dependent on 2 bobs being resolved: BOB-30182 and BOB-248

**Comment by 317brian** [comment]
@dogonthehorizon to review this and then we can merge

File: content/en/docs/spinnaker-user-guides/using-dinghy.md [hunk 1]
```diff
@@ -60,6 +60,9 @@ Here is an example Dinghyfile:
 ```json
 {
   "application": "helloworldapp",
+  "appmetadata": {
+    "canaryConfigs": "true"
+  },
```
Comment (cristhian-castaneda) [inline content/en/docs/spinnaker-user-guides/using-dinghy.md]: This suggestion adds a few more configuration options for `appmetadata` 

```suggestion
  "globals": {
    "save_app_on_update": true
  },
  "spec": {
    "appmetadata": {
      "dataSources": {
        "enabled": [
          "securityGroups",
          "loadBalancers",
          "executions",
          "serverGroups"
        ],
        "disabled": [
          "canaryConfigs"
        ]
      },
      "cloudProviders": [
        "kubernetes",
        "aws"
      ],
      "customBanners": [
        {
          "backgroundColor": "var(--color-alert)",
          "enabled": true,
          "text": "Your custom banner text",
          "textColor": "var(--color-text-on-dark)"
        }
      ]
    }
  },
```

File: content/en/docs/spinnaker-user-guides/using-dinghy.md [hunk 2]
```diff
@@ -79,6 +82,7 @@ Here is an example Dinghyfile:
 Make sure you specify the following fields so that the Dinghyfile can create a pipeline with stages:
 
 * `.application`: The name of the application where pipelines will be created or updated.  If the application does not exist, it will be created.
+* `appmetadata`: Key/value pairs where you can define application metadata, such as whether Canary analysis is enabled (`"canaryConfigs": "true"`). You can retrieve a list of potential configs by opening the developer console for your browser when viewing the **Config** page of an application. Look for the possible parameters under the `spinnaker.application.attributes` key. Note that no validation is done for this field.
```
Comment (cristhian-castaneda) [inline content/en/docs/spinnaker-user-guides/using-dinghy.md]: This suggestion adds a few more descriptions for configuration options on `appmetadata`

```suggestion
* `.globals`: Configuration used to define top-level variables. See [top-level variables](#top-level-variables) for more information.
* `.globals.save_app_on_update`:  Flag that allows updates to application properties. Ssee [see Application updates](#application-updates) for more information.
* `.appmetadata`: Key/value pairs where you can define application metadata, such as whether Canary analysis is enabled. You can retrieve a list of potential configs by opening the developer console for your browser when viewing the **Config** page of an application. Look for the possible parameters under the `spinnaker.application.attributes` key. Note that no validation is done for this field.
* `.appmetadata.dataSources.enabled`:  An array of features that will be displayed on Spinnaker
* `.appmetadata.dataSources.disabled`: An array of features that will not be displayed on Spinnaker
* `.appmetadata.customBanners[*]`: An array of banners that will be displayed on Spinnaker, must only have one item checked as enabled
* `.appmetadata.cloudProviders[*]`: An array of cloud providers for the application```

**Review by cristhian-castaneda — APPROVED** [review]
LGTM

**Pull request merged**